src/core/createInstance.js

```diff
@@ -43,48 +42,59 @@ export default (
     }
     const config = createConfig(options);
 
-    try {
-      componentRegistry = initializeComponents(config);
-    } catch (e) {
-      configurationFailed = true;
-      throw e;
-    }
+    return initializeComponents(config);
   };
 
   const executeCommand = (commandName, options) => {
-    let command;
+    let execute;
 
-    if (configurationFailed) {
-      // We've decided if configuration fails we'll return
-      // never-resolved promises for all subsequent commands rather than
-      // throwing an error for each of them.
-      command = () => new Promise(() => {});
-    } else if (commandName === "configure") {
-      if (componentRegistry) {
+    if (commandName === "configure") {
+      if (configurePromise) {
         throw new Error(
           "The library has already been configured and may only be configured once."
         );
       }
-      command = configureCommand;
-    } else if (commandName === "log") {
-      command = logCommand;
+
+      execute = () => {
+        configurePromise = configureCommand(options);
+        return configurePromise;
+      };
     } else {
-      if (!componentRegistry) {
+      if (!configurePromise) {
         throw new Error(
           `The library must be configured first. Please do so by calling ${namespace}("configure", {...}).`
         );
       }
 
-      command = componentRegistry.getCommand(commandName);
-
-      if (!isFunction(command)) {
-        throw new Error(`The ${commandName} command does not exist.`);
+      if (commandName === "log") {
```

Comment:
This change requires that configure be called before "log".  Do we want that?

Reply:
I think so. I like the blanket statement of saying you have to run `configure` before anything else. I think I like this because it doesn't run the risk of being a slippery slope of what may or may not be supported before configure and it makes it pretty clear that `configure` lays the foundation for settings and then other commands apply on top of that. For example, if I were to have:

```
alloy("log", {
  enabled: true
});

alloy("configure", {
  logEnabled: false
});
```

It's just a little odd to me and might result in confusion as to whether logging will ultimately be enabled. Because `configure` to me sounds like an initialization/foundational step, this seems like a more understandable approach:

```
alloy("configure", {
  logEnabled: false
});

alloy("log", {
  enabled: true
});
```

Anyway, that's my thought behind it. Thanks for bringing it up.